tests/test_watson.py

```diff
@@ -794,6 +794,12 @@ def test_report(watson):
     with pytest.raises(WatsonError):
         watson.report(arrow.now(), arrow.now(), tags=["A"], ignore_tags=["A"])
 
+    watson.start('bar')
+    _ = watson.report(arrow.now(), arrow.now(), current=True)
+    watson.stop()
+
+    assert watson.frames['id'].count('current') == 0
```

Comment:
It looks like you're testing the implementation here (i.e. the current frame has been deleted after report() finishes), but maybe it would be even more useful to test the actual report data returned?

The question I'd like to get answered by the test is this: _are we sure that the current frame is not being deleted too soon in the report() method?_

The `watson.py` implementation looks fine to me :+1:

Reply:
@jmkerr I'm not sure if I made myself clear. If you need further clarification just let me know.

@k4nar Sorry for stepping in here, I thought it might be useful to move the PR forward... what do you think about my comment?

Reply:
I'll try to explain my point further: I think you should solely rely on `Watson` class methods to test your assumptions on the behavior of the `Watson` class.

As I see it, by accessing the `Frames` object and asserting on its properties, you're effectively testing an implementation detail of the `Watson` class and adding complexity to the test itself.

What I would have done here is calling two or more times `watson.report()` and verify that you're always getting the expected report results.

What do you think? :relaxed: